src/QsFmt/App/Program.fs

```diff
@@ -106,20 +129,174 @@ let runUpdateAndFormat (arguments: UpdateAndFormatArguments) =
 
     match Arguments.fromUpdateArguments asUpdateArguments with
     | Ok args -> args.Input |> run { args with CommandKind = UpdateAndFormat }
-    | Error errorCode -> errorCode
+    | Error errorCode -> { RunResult.Default with ExitCode = errorCode }
+
+type ParseArgsResult =
+    | Success of ParserResult<obj>
+    | Error of Exception
+
+type ExecutionResult =
+    | Success of RunResult
+    | Error of Exception
```

Comment:
You could use `Result<ParserResult<obj>, Exception>` and `Result<RunResult, Exception>` instead (`Result` is a built-in F# type with constructors `Ok` and `Error`).

Reply:
Suggestion implemented.